src/osd/OSD.cc

```diff
@@ -9485,9 +9485,9 @@ void OSDShard::consume_map(
 	slot->waiting.pop_front();
       }
     }
-    if (slot->waiting.empty() &&
-	slot->num_running == 0 &&
-	!slot->pg) {
+    if (!slot->pg &&
```

Comment:
@majianpeng why would you think it improves the performance? could you elaborate a little bit in the commit message?

Reply:
quote of the commit message
> For conditons "slot->num_running == 0 && slot->waitting.empty()" also
mean there is no ops. But !slot->pg mean it want to be removed after
_detach_pg. So firstly check !slot->pg.

i still don't get why this implies better performance. because checking `slot->pg` would be fast than checking `slot->waiting.empty()` plus `slot->num_running == 0` ?

instead, i think it's better to check `slot->waiting.empty()` first here. as this function is working around updating the work items attached to slots in `pg_slots` when new osd_map is consumed. so it's natural to check it first.

do you have any statistics supporting this change?